packages/surveys/src/components/general/survey.tsx

```diff
@@ -132,9 +132,23 @@ export function Survey({
   const [localSurvey, setlocalSurvey] = useState<TJsEnvironmentStateSurvey>(survey);
   const [currentVariables, setCurrentVariables] = useState<TResponseVariables>({});
 
+  // state to keep track of the original required states of the questions
+  const originalQuestionRequiredStates = useRef<Record<string, boolean>>({});
+
+  // state to keep track of the questions that were made required by each specific question's logic
+  const questionRequiredByMap = useRef<Record<string, string[]>>({});
+
   // Update localSurvey when the survey prop changes (it changes in case of survey editor)
   useEffect(() => {
     setlocalSurvey(survey);
+    // Update original required states when survey changes
+    originalQuestionRequiredStates.current = survey.questions.reduce<Record<string, boolean>>(
```

Comment:
I think changing this here in the useEffect now kind of defeats the purpose of defining this as a ref with `useRef`. I think we can just define this as a value returned from `useMemo` in which the deps array also contains the `survey` object 🙏